server/src/main/java/org/opensearch/action/admin/cluster/shards/CatShardsRequest.java

```diff
@@ -0,0 +1,76 @@
+/*
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * The OpenSearch Contributors require contributions made to
+ * this file be licensed under the Apache-2.0 license or a
+ * compatible open source license.
+ */
+
+package org.opensearch.action.admin.cluster.shards;
+
+import org.opensearch.action.ActionRequestValidationException;
+import org.opensearch.action.support.clustermanager.ClusterManagerNodeReadRequest;
+import org.opensearch.common.unit.TimeValue;
+import org.opensearch.core.common.io.stream.StreamInput;
+import org.opensearch.core.tasks.TaskId;
+import org.opensearch.rest.action.admin.cluster.ClusterAdminTask;
+
+import java.io.IOException;
+import java.util.Map;
+
+import static org.opensearch.search.SearchService.NO_TIMEOUT;
+
+/**
+ * A request of _cat/shards.
+ *
+ * @opensearch.api
+ */
+public class CatShardsRequest extends ClusterManagerNodeReadRequest<CatShardsRequest> {
```

Comment:
Instead of extending from ClusterManagerNodeReadRequest which should be used when extending TransportClusterManagerNodeAction, you can create AbstractCatRequest and keep common params in the abstract class

Reply:
In CatShards request we are setting few parameters which are used in Cluster state request that's why we need to extend ClusterManagerNodeReadRequest.

Reply:
you are anyway duplicating  clusterManagerNodeTimeout, local, dont understand the benefit of extending ClusterManagerNodeReadRequest

Reply:
In rest shards action we fetch default value of manager node timeout that's why we need to extend ClusterManagerNodeReadRequest.

Reply:
then why are you duplicating the field?

Reply:
Ack. Removing getter setter for clusterManagerNodeTimeout.

Reply:
local field as well

Reply:
> In rest shards action we fetch default value of manager node timeout that's why we need to extend ClusterManagerNodeReadRequest.
If you are extending only for this. you shouldn't you can access it directly as ```ClusterManagerNodeRequest.DEFAULT_CLUSTER_MANAGER_NODE_TIMEOUT```

Reply:
discussed offline to extend the ClusterManagerNodeRequest and try to move commen things.